boltdesignsystem/bolt
# [6.x] DS-855 Prep Bolt in Drupal

## Jira

https://pegadigitalit.atlassian.net/browse/DS-855

## Summary

Prepare Bolt to be moved into our Drupal base theme.

## Details

- Setup React as external dependency, i.e. it will be on the window not bundled with Webpack
  - Required adding CDN link to React in head of PL demos, test environments, and microjourneys editor iframe
- Fix mismatched dependency warnings
  - Required upgrading `react-html-parser`
- Add `async` option to `bolt` cli, required by Drupal
- Move drupal-specific configuration directly into Bolt
- Enable `watch` command to work in Drupal environment
- Lock `scroll-js` to v3.1.1 to prevent Node version error

## How to test

- Review code
- Verify Bolt commands work as expected (start, build, etc)

To test this PR in Drupal you must have Saleshub setup locally. For now just review this code in Bolt and verify there are no regressions to Bolt. Any additional changes required for Drupal will be handled in future PRs.

## Release notes

**Breaking Change**
Microjourneys editor will require React to be passed in via the `scripts` prop. See [example in PR](https://github.com/boltdesignsystem/bolt/pull/2578/files#diff-c7bb7897e243c1a197a0829c37d1ddbed0fd71db50bf2ac2e4999e34ff7e6cd1R4). This loads React in the grapes JS editor iframe. It is required because bolt-global.js is loaded in the iframe and bolt-global.js expects React to be on the window.
